utils.go

```diff
@@ -17,7 +17,14 @@ func isBtrfs(path string) (bool, error) {
 	if err := syscall.Statfs(path, &stfs); err != nil {
 		return false, &os.PathError{Op: "statfs", Path: path, Err: err}
 	}
-	return int64(stfs.Type) == SuperMagic, nil
+	fsType := int64(stfs.Type)
+	if unsafe.Sizeof(stfs.Type) == 4 {
+		// We're on a 32 bit arch, where statfs.Type is int32.
+		// The expected value is negative number when interpreted
+		// as int32 so we need to cast via uint32 to avoid sign extension.
+		fsType = int64(uint32(stfs.Type))
+	}
+	return fsType == SuperMagic, nil
```

Comment:
The `SuperMagic` is a constant which is already 32 bit, so we could cast to `uint32` on both platforms, I think.

Just in case, we can set type for `SuperMagic` explicitly as well (`const SuperMagic uint32 = ...`). In case it will differ on some 64 bit platform, Go will catch the size mismatch.

Reply:
will try